webknossos/__generate_client.py

```diff
@@ -0,0 +1,121 @@
+import calendar
+import json
+from datetime import datetime
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+from typing import Dict, Iterable, Tuple
+
+import httpx
+from inducoapi import build_openapi
+from openapi_python_client import Config, MetaType, _get_project_for_url_or_path
+from openapi_python_client.cli import handle_errors
+
+from webknossos.client import _get_generated_client
+
+SCHEMA_URL = "https://converter.swagger.io/api/convert?url=https%3A%2F%2Fwebknossos.org%2Fswagger.json"
+
+
+def assert_valid_schema(openapi_schema: Dict):
+    assert openapi_schema["openapi"].startswith("3.0.")
+
+
+def generate_client(openapi_schema: Dict):
+    assert_valid_schema(openapi_schema)
+    with NamedTemporaryFile("w", suffix=".json") as schema_file:
+        schema_file.write(json.dumps(openapi_schema))
+        generator_config = Config(
+            project_name_override="webknossos/client/generated",
+            package_name_override=".",
+        )
+        generator_project = _get_project_for_url_or_path(
+            url=None,
+            path=Path(schema_file.name),
+            meta=MetaType.POETRY,
+            config=generator_config,
+        )
+        errors = generator_project.update()
+        # handle_errors(errors)  # prints warnings
```

Comment:
remove comment?

Reply:
This might help if the generation fails somehow, therefore I'd leave it in.